frappe/model/document.py

```diff
@@ -586,6 +587,17 @@ def set_docstatus(self):
 		for d in self.get_all_children():
 			d.docstatus = self.docstatus
 
+	def set_tracer(self):
+		if (
+			getattr(self.meta, "trace_flow", False)
+			and not getattr(self.meta, "issingle", False)
+			and self.is_new()
+			and self.tracer is None
+		):
+			tracer = frappe.new_doc("Flow Tracer")
+			tracer.db_insert()
+			self.tracer = tracer.name
+
```

Comment:
Inside a request, all initiation tracable entry-documents should probably derive their tracer predictably from the `__frappe_exc_id` so that this request association can be encoded into the (each?) tracer for later recovery and association.